Sources/API/APIHeaders.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  APIHeaders.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public typealias HTTPHeaders = [String: String]
+
+public struct APIHeaders {
+    
+    public typealias UserAgent = String
+    private static var userAgent: UserAgent?
+    public static func setUserAgent(_ userAgent: UserAgent) {
+        self.userAgent = userAgent
+    }
+    
+    public init() { }
+    
+    public var defaultHeaders: HTTPHeaders {
+        guard let userAgent = APIHeaders.userAgent else {
+            fatalError("Please set the userAgent before accessing defaultHeaders.")
```

Comment:
This is time-sensitive, based on when `setUserAgent ` is called, right? Shouldn't it rather be an assertion, to not crash the app on production?